src/Sinch/Conversation/Messages/Send/SendMessageRequest.cs

```diff
@@ -154,11 +157,34 @@ public override string ToString()
             sb.Append("  ConversationMetadata: ").Append(ConversationMetadata).Append("\n");
             sb.Append("  Queue: ").Append(Queue).Append("\n");
             sb.Append("  Recipient: ").Append(Recipient).Append("\n");
-            sb.Append("  Ttl: ").Append(Ttl).Append("\n");
+            sb.Append("  Ttl: ").Append(TtlSeconds).Append("\n");
             sb.Append("  ProcessingStrategy: ").Append(ProcessingStrategy).Append("\n");
             sb.Append("  CorrelationId: ").Append(CorrelationId).Append("\n");
             sb.Append("}\n");
             return sb.ToString();
         }
     }
+
+    public class TimeToLiveConverter : JsonConverter<int?>
+    {
+        public override int? Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            var str = reader.GetString();
+            if (str is null) return null;
+            str = str.Trim();
+            var val = new string(str.TakeWhile(char.IsNumber).ToArray());
+            if (int.TryParse(val, out int result))
+            {
+                return result;
+            }
+
+            return null;
+        }
+
+        public override void Write(Utf8JsonWriter writer, int? value, JsonSerializerOptions options)
```

Comment:
Can you also add a test for this one, in order to cover the case where the TTL is null (the e2e only covers the nominal case)? And maybe check if the value is in the correct range.

Reply:
ttl null is checked in tests above as well, `WithJson` matcher makes a deep equal comparison, so if `ttl` is present, there will be an error. And any extended validation for other fields then obvious missing id is not supported by sdk

Reply:
Let's keep it this way then